packages/main-nav/src/MainNavItem.tsx

```diff
@@ -1,79 +1,72 @@
-import { ReactNode, ButtonHTMLAttributes } from 'react';
-import { Flex, BoxProps, Box } from '@ag.ds-next/box';
+import {
+	ReactNode,
+	ButtonHTMLAttributes,
+	PropsWithChildren,
+	ElementType,
+} from 'react';
+import { Flex, Box } from '@ag.ds-next/box';
 import {
 	boxPalette,
 	LinkProps,
 	useLinkComponent,
-	forwardRefWithAs,
 	mapSpacing,
 	mq,
 } from '@ag.ds-next/core';
 import { localPalette } from './utils';
 
-const MainNavItem = forwardRefWithAs<'div', BoxProps>(function MainNavItem({
-	as = 'div',
-	...props
-}) {
-	return (
-		<Box
-			paddingBottom={0.5}
-			height="100%"
-			css={mq({
-				' a, button': {
-					position: 'relative',
-					color: boxPalette['foregroundAction'],
-					padding: mapSpacing(1),
-					'&:hover': {
-						color: boxPalette.foregroundText,
-						backgroundColor: localPalette.linkHoverBg,
-					},
-				},
-				' svg': {
-					height: 24,
-					width: 24,
+type MainNavItemProps = PropsWithChildren<{
+	as: ElementType;
+}>;
+
+const MainNavItem = ({ as, ...props }: MainNavItemProps) => (
```

Comment:
nitpick here, but I prefer adding `{ return(` to the components, as It makes the diff easier later on if we ever had to add more logic to components. What do you think?